zookeeper-server/src/main/java/org/apache/zookeeper/server/Request.java

```diff
@@ -164,6 +169,26 @@ public boolean isThrottlable() {
                 && this.type != OpCode.createSession;
     }
 
+    private transient byte[] serializeData;
+
+    @SuppressFBWarnings(value = "EI_EXPOSE_REP")
+    public byte[] getSerializeData() {
+        if (this.hdr == null) {
+            return null;
+        }
+
+        if (this.serializeData == null) {
+            try {
+                this.serializeData = Util.marshallTxnEntry(this.hdr, this.txn, this.txnDigest);
+            } catch (IOException e) {
+                LOG.error("This really should be impossible.", e);
+                this.serializeData = new byte[32];
```

Comment:
I think this is a bad idea even if it shouldn't happen. For unhandled IOExceptions, I think that a good approach is to wrap them with `java.io.UncheckedIOException`. 
For example, in this case:
`throw new UncheckedIOException(e);` (also add `import java.io.UncheckedIOException`)

Reply:
Make sense.